artifactory/utils/buildutils_test.go

```diff
@@ -68,7 +69,39 @@ func TestGetBuildName(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, buildNameFile, actualBuildName)
 }
+func TestGetEmptyBuildNameOnUnixAccessDenied(t *testing.T) {
+	if coreutils.IsWindows() {
+		t.Skip("Skipping TestGetEmptyBuildNameOnUnixAccessDenied test on windows...")
+	}
+	// Create build config in temp folder.
+	tmpDir, createTempDirCallback := tests.CreateTempDirWithCallbackAndAssert(t)
+	defer createTempDirCallback()
+
+	// Create build config in temp folder
+	destConfFile := filepath.Join(tmpDir, ".jfrog", "projects")
+	srcConfFile := filepath.Join("testdata", "build.yaml")
+	assert.NoError(t, fileutils.CopyFile(destConfFile, srcConfFile))
+
+	// Remove permissions from config file
+	err := os.Chmod(destConfFile, 0000)
+	assert.NoError(t, err)
+
+	// Validate build name form config file doesn't throw an error if access is denied.
+	wd, err := os.Getwd()
+	assert.NoError(t, err, "Failed to get current dir")
+	chdirCallBack := testsutils.ChangeDirWithCallback(t, wd, tmpDir)
+	defer chdirCallBack()
+	buildConfig := NewBuildConfiguration("", "", "", "")
+	actualBuildName, err := buildConfig.GetBuildName()
+	assert.NoError(t, err)
+	assert.False(t, buildConfig.loadedFromConfigFile)
+	assert.Empty(t, actualBuildName)
 
+	// Restore permissions for deleting the config file.
+	err = os.Chmod(destConfFile, 0770)
```

Comment:
This should be probably done in a defer clause, right after the initial permission change.
The way it is right now, if an error occurs after the initial permission change, the deletion of the temp file will fail. Actually, it is best to put this right before the code that deletes the file.